lib/rust/ensogl/core/src/display/shape/primitive/shader/canvas.rs

```diff
@@ -37,27 +37,14 @@ impl Shape {
 /// reference to GLSL code which defines a vector shape there.
 #[derive(Clone, Debug)]
 pub struct ShapeData {
-    shape_num: usize,
-    ids:       Vec<usize>,
-    name:      String,
+    name: String,
 }
 
 impl ShapeData {
     /// Constructor.
-    pub fn new(shape_num: usize) -> Self {
-        let ids = default();
-        let name = format!("shape_{}", shape_num);
-        Self { shape_num, ids, name }
-    }
-
-    /// Adds new id enclosed in this shape.
-    pub fn add_id(&mut self, id: usize) {
```

Comment:
This is some big change!
1. There is no explanation in the PR why it is done.
2. There is no explanation how it influences the program behavior.
3. I am not sure this is correct now. Anyway, the most important question - why this change was needed and not described?

Reply:
The whole `ids` vec was simply unused. The only usages were writes or copies from one into another. I simply removed dead code here, also removing some unnecessary vec allocations from shader generator.

About the shape naming change, it used to be based on pointer addresses of temporary values, which were effectively nondeterministic. Each time a shader was built using the same shape definition, the generate code was different due to those names being different. That meant that the shader cache generated different content hash every time. Using a sequential numbering for names fixes that completely. Interestingly, the machinery for that sequential numbering was already in place. The shape deduplication logic still uses the same pointer values for comparison. The only change in shader code output is numeric parts of shape function names.

I'm going to add a comment to the code that assigns the IDs to make it clear what property we expect there.

Reply:
Ok, that makes sense. WOW, I did not realise ids were not used. Hmmm, ok, thanks for clarification. I'd not write this comment if I knew about it.